src/connector.js

```diff
@@ -0,0 +1,612 @@
+	if (!statusController.state.firstSearchExecuted && !hashParams.q ) {					
+		return window.location.search.slice( 1 ); // use query string if hash is empty
+	}
+	
+	return hash;
+};
+
+const urlManager = buildUrlManager( headlessEngine, {
+	initialState: {
+		fragment: fragment(),
+	},
+} );
+
+// Sync controllers when URL changes
+const onHashChange = () => { 
+    urlManager.synchronize( fragment() );
+};
+
+// Execute a search if parameters in the URL on page load
+if ( !statusController.state.firstSearchExecuted && fragment() ) {
+	headlessEngine.executeFirstSearch();
+}
+
+// Listen to URL change (hash)
+window.addEventListener( 'hashchange', onHashChange );
+
+// Unsubscribe to controllers
+const unsubscribeManager = urlManager.subscribe( () => {
+	if ( !params.enableHistoryPush	|| window.location.origin.startsWith( 'file://' ) ) {
+		return;
+    }
+		
+	let hash = `#${urlManager.state.fragment}`;
+
+	if ( window.location.href.includes( '#' ) ) {
+		hash = hash.replace( '#', '&' );
+    }
+	
+	if ( !statusController.state.firstSearchExecuted ) {
+		window.history.replaceState( null, document.title, window.location.href + hash );
+		return;
+	}
+	window.history.pushState( null, document.title, window.location.href + hash );
+});
+
+let searchBoxState = null;
+let resultListState = null;
+let querySummaryState = null;
+let pagerState = null;
+
+let lastCharKeyUp = null;
+
+// Show query suggestions if a search action was not executed (if enabled)
+function updateSearchBoxState( newState ) {
+	searchBoxState = newState;
+	if ( !suggestionsElement ) {
+		return;
+    }
+		
+	if ( lastCharKeyUp == 13 ) {
+		suggestionsElement.style.display = 'none';
+		return;
+	}
+
+	if ( !searchBoxState.isLoadingSuggestions ) {
+		suggestionsElement.textContent = '';
+		searchBoxState.suggestions.forEach( ( suggestion ) => {
+			const node = document.createElement( "li" );
+			node.setAttribute( "class", "suggestion-item" );
+			node.onclick = ( e ) => { 
+                searchBoxController.selectSuggestion(e.currentTarget.innerText);
+                searchBoxElement.value = e.currentTarget.innerText;
+            };
+			node.innerHTML = suggestion.highlightedValue;
+			suggestionsElement.appendChild( node );
+		});
+		
+		if ( searchBoxState.suggestions.length > 0 ) {
+			suggestionsElement.style.display = 'block';
+		}
+	}
+}
+
+// Filters out dangerous URIs that can create XSS attacks such as `javascript:`.
+function filterProtocol( uri ) {
+
+	const isAbsolute = /^(https?|ftp|file|mailto|tel):/i.test( uri );
+	const isRelative = /^(\/|\.\/|\.\.\/)/.test( uri );
+
+	return isAbsolute || isRelative ? uri : '';
+}
+
+// Clear results list
+function clearResultList() {
+	resultListElement.textContent = "";
+}
+
+// Update results list
+function updateResultListState( newState ) {
+	resultListState = newState;
+	
+	if ( resultListState.isLoading ) {
+		if ( suggestionsElement ) {
+			suggestionsElement.style.display = 'none';
```

Comment:
Considering there is a formal function to show it, let use a formal function to hide it too.

```suggestion
			searchBoxController.hideSuggestions();
```

Reply:
This is a note for @bblaisATcoveo for his subsequent PR